### DISCUSSION TITLE
navigate to a new router with snapshot state is fail

### DISCUSSION THREAD

**Discussion by Pulset** [pmndrs/valtio#841] [Bug report]
I navigate to a new router with snapshot state,but it's fail.With a normal object is successful.
demo:https://codesandbox.io/p/sandbox/interesting-dust-jkzw79?file=%2Fsrc%2FApp.js%3A7%2C4

**Comment by dai-shi** [comment]
Hmm, it doesn't seem like it accepts snapshot object.
Would you like to try deep cloning the snapshot?

**Reply by Pulset** [reply]
why it can't accepts snapshot object?

**Reply by dai-shi** [reply]
I actually don't know. snapshot object returned by `useSnapshot` is somewhat special with another proxy, so some libraries don't like it.

**Reply by Pulset** [reply]
Thanks
